pkg/metrics/memory_linux.go

```diff
@@ -109,23 +122,30 @@ parse:
 		case "MemAvailable":
 			ret.Available = t * 1024
 			memAvailable = true
-			if memTotal {
-				break parse // stop parsing if we have enough fields
-			}
 		case "MemTotal":
 			ret.Total = t * 1024
-			memTotal = true
-			if memAvailable {
-				break parse // stop parsing if we have enough fields
-			}
+			readFields++
 		case "MemFree":
 			ret.Free = t * 1024
+			readFields++
 		case "Buffers":
 			ret.Buffers = t * 1024
+			readFields++
 		case "Cached":
 			ret.Cached = t * 1024
+			readFields++
+		case "Shmem":
+			ret.Shared = t * 1024
+			readFields++
+		case "Slab":
+			ret.Slab = t * 1024
+			readFields++
 		case "SReclaimable":
 			ret.SReclaimable = t * 1024
+			readFields++
+		}
+		if readFields >= 7 && memAvailable { // stop reading the file when we have read all the fields we require
```

Comment:
why not increase readFields on `memAvailable` and just compare readFields ?

Reply:
memAvailable is expected to miss on some kernels.